### PR TITLE
add `D.Multinomial`

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**

### CODE REVIEW

**Comment by takagi** [comment]
`scipy.stats.multinomial` is supported only SciPy 0.19 and later. Is it possible to use `numpy.random.multinomial` instead?

**Comment by toslunar** [comment]
`scipy.stats.multinomial` is used only in the tests.  I'd fix it by adding `>=0.19` to the line `@with_requires('scipy')`.

**Comment by takagi** [comment]
It makes sense.

File: tests/chainer_tests/distributions_tests/test_multinomial.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+
+@testing.parameterize(*testing.product({
+    'shape': [(2, 3), ()],
+    'is_variable': [True, False],
+    'sample_shape': [(3, 2), ()],
+    'k': [3],
+}))
+@testing.fix_random()
+@testing.with_requires('scipy')
```
Comment (takagi) [inline tests/chainer_tests/distributions_tests/test_multinomial.py]: @YoshikawaMasashi Would you apply the following change?
```suggestion
@testing.with_requires('scipy>=0.19')
```

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by YoshikawaMasashi** [comment]
thanks for the fix.

**Comment by takagi** [comment]
@toslunar Would you also review this PR because I'm not with expertise in probability and statistics.

File: chainer/distributions/multinomial.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        obo_n = xp.broadcast_to(
+            self.n.data, self.batch_shape).reshape(-1)
+        if xp == cuda.cupy:
+            obo_n = obo_n.get()
+        print([(one_n, one_p) for one_n, one_p in zip(obo_n, obo_p)])
+        eps = [xp.random.multinomial(
+            one_n, one_p, size=n) for one_n, one_p
+            in zip(obo_n, obo_p)]
+        eps = xp.stack(eps)
+        eps = xp.swapaxes(eps, 0, 1)
```
Comment (toslunar) [inline chainer/distributions/multinomial.py]: `stack` is not available in `numpy<1.10`.  How about using `xp.expand_dims(..., axis=1)` and `xp.concatenate`.

File: chainer/distributions/multinomial.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        return self.n * self.p
+
+    def sample_n(self, n):
+        xp = cuda.get_array_module(self.p)
+        obo_p = self.p.data.reshape(-1, self.p.shape[-1])
+        obo_n = xp.broadcast_to(
+            self.n.data, self.batch_shape).reshape(-1)
+        if xp == cuda.cupy:
+            obo_n = obo_n.get()
+        print([(one_n, one_p) for one_n, one_p in zip(obo_n, obo_p)])
```
Comment (toslunar) [inline chainer/distributions/multinomial.py]: Would you remove this line?

**Pull request closed without merging**